core/src/main/java/org/apache/iceberg/jdbc/JdbcClientPool.java

```diff
@@ -48,7 +48,12 @@
   protected Connection newClient() {
     try {
       Properties dbProps = new Properties();
-      properties.forEach((key, value) -> dbProps.put(key.replace(JdbcCatalog.PROPERTY_PREFIX, ""), value));
+      properties.forEach((key, value) -> {
+        if (key.startsWith(JdbcCatalog.PROPERTY_PREFIX)) {
+          String dbKey = key.replaceFirst("^" + JdbcCatalog.PROPERTY_PREFIX, "");
```

Comment:
Nit / non-blocking: Can you possibly abstract this into a utility method in maybe a `JdbcCatalogUtil` class or something similar?

The need to iterate over properties, filter the ones that don't match the prefix, and then update the keys, is usually pretty common.

Reply:
Also, could this be simplified to not use the regex?

Like
```java
// We might need to check that its got the name in it too (e.g. that it doesn't belong to another catalog).
private void jdbcPropertyPrefixForCatalog(String name) {
  return String.format("%s"."%s", JdbcCatalog.PROPERTY_PREFIX, name);
}
properties.forEach((key, value) -> {
  if (key.startsWith(JdbcCatalog.PROPERTY_PREFIXdbProps.put(key.replace(JdbcCatalog.PROPERTY_PREFIX, ""), value));
```

Reply:
Also, could this be simplified to not use the regex? I looked at some of the surrounding class and it does seem they use a regex, so maybe it's the string concatenation that's getting to me?

Also, do we need to strip more than just `jdbc` from the string at this point? And do we need to check that these properties are specifically for this JDBC catalog?

For example, if I were using spark-sql and I had two catalogs, `spark.sql.catalog.jdbc_1` and `spark.sql.catalog.jdbc_2`, should we do a name check here or is that already handled? And is there any chance the regex will do something we don't intend for it to do if the catalog is named `jdbc_foo`? I'm guessing not because of `replaceFirst`.

Either way, this looks much better to me.

**TLDR** - Can you add test cases around configuring the right jdbc user - possibly in `TestJdbcCatalog` (or say, from Spark or from Flink if it's hard to test from there for some reason)?